### ISSUE TITLE
Inconsistency detected by ld.so

### ISSUE THREAD

**Issue by jaredlander** [mlverse/torch#663]
When running `torch::install_torch()` in a project using `{renv}` I'm getting the following error.

```r
> torch::install_torch(type='11.1')
Inconsistency detected by ld.so: ../sysdeps/x86_64/dl-machine.h: 540: elf_machine_rela_relative: Assertion `ELFW(R_TYPE) (reloc->r_info) == R_X86_64_RELATIVE' failed!
```

I don't seem to have this problem if installing torch for the CPU. And this error only appears if running R in the terminal, otherwise it crashes RStudio Workbench.

The error is caused even by typing `torch::install_t` then hitting `<tab>` for autocompletion or running `library(torch)`. And it always crashes me out of R.

To double check, I'm running CUDA 11.1.

```sh
nvcc --version
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2020 NVIDIA Corporation
Built on Mon_Oct_12_20:09:46_PDT_2020
Cuda compilation tools, release 11.1, V11.1.105
Build cuda_11.1.TC455_06.29190527_0
```

I have removed CUDA, cudnn8 and all Nvidia drivers and reinstalled them.

I also checked that CUDA works by compiling the samples in `/usr/local/cuda-11.1/samples`.

Not sure what this message means at this point.

```r
> sessionInfo()
R version 4.1.0 (2021-05-18)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 18.04.5 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas/libblas.so.3
LAPACK: /usr/lib/x86_64-linux-gnu/libopenblasp-r0.2.20.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C
 [9] LC_ADDRESS=C               LC_TELEPHONE=C
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C

attached base packages:
[1] stats     graphics  grDevices datasets  utils     methods   base

loaded via a namespace (and not attached):
[1] compiler_4.1.0 tools_4.1.0    renv_0.13.0
```

**Comment by jaredlander** [comment]
For what it's worth, when not using `{renv}` I can install torch and `torch::cuda_device_count()` shows that I have a GPU. So it might be `{renv}` not playing nicely with the installer.

**Comment by dfalbel** [comment]
I couldn't reproduce this yet, but I think the problem could be that you have this same version of torch installed in a different `renv` project but using the CPU (or a different version of CUDA). Since `renv` is using a cache directory to store the packages and we install LibTorch in the same folder as the package (by default), that might be causing the inconsistency.

I think you can solve this by erasing the `deps` folder here: `list.files(file.path(system.file(package="torch"), "deps"))` and just trying to load `torch` again - that should trigger the auto-installation of LibTorch and now the correct version should be installed.

Another alternative is setting the `TORCH_HOME` environment variable somewhere inside the project folder, so each project can have its own LibTorch installation.

For example, you can add this: https://github.com/mlverse/torch-learnr/blob/master/.Rprofile#L2 to your .Rprofile file.

**Comment by jaredlander** [comment]
So theoretically, by setting the TORCH_HOME environment variable, I could have multiple versions of torch, and some projects would use one version and other projects would use other versions?

**Comment by dfalbel** [comment]
Yes! Note that technically speaking its multiple flavors of LibTorch. Each {torch} pkg version will have a specific requirement for the LibTorch version (eg 0.4 uses libtorch 1.8, 0.5 uses libtorch 1.9). However LibTorch can have multiple flavors (eg: CPU, CUDA10, CUDA11). 
With TORCH_HOME, you can choose which flavor of LibTorch the R package will use for each {renv} project.

**Comment by jaredlander** [comment]
That's great, I'm going to try that, thanks.

Speaking of flavors, do you know when CUDA > 11.1 will be supported?

**Comment by dfalbel** [comment]
Great! Let me know if this works.

No idea, I think LibTorch already supports CUDA 11.3 (See eg https://github.com/pytorch/pytorch/issues/56721) but they are still not providing pre-built binaries for 11.3.

**Comment by dfalbel** [comment]
11.3 is now supproted with the dev version #758 and #763
